opencga-analysis/src/main/java/org/opencb/opencga/analysis/rga/RgaManager.java

```diff
@@ -1952,4 +1991,57 @@ public Preprocess setEvent(Event event) {
             return this;
         }
     }
+
+    /*
+    CACHE METHODS
+     */
+    private String generateCacheKey(String method, String studyStr, Query query, QueryOptions options) {
+        ObjectMap map = new ObjectMap()
+                .append("method", method)
+                .append("study", studyStr);
+        if (query != null) {
+            map.putAll(query);
+        }
+        if (options != null) {
+            map.putAll(options);
+        }
+        // Sort the keys
+        List<String> sortedKeys = map.keySet().stream().sorted().collect(Collectors.toList());
+        List<String> queryList = new ArrayList<>(map.size());
+        for (String key : sortedKeys) {
+            queryList.add(key + "=" + map.get(key));
+        }
+        return DigestUtils.sha256Hex(StringUtils.join(queryList, ";"));
```

Comment:
For this kind of "hashes" that are purely for ensuring having a unique string, it's preferrable md5 over sha.

https://www.geeksforgeeks.org/difference-between-md5-and-sha1/


> Both MD5 stands for Message Digest and SHA1 stands for Secure Hash Algorithm square measure the hashing algorithms wherever The speed of MD5 is fast in comparison of SHA1’s speed.

Reply:
In this case, it could be even "enough" to use the plain String.hashCode, or not use a hash at all, and relay on the HashMap itself.

```java
return StringUtils.join(queryList, ";");
```

Reply:
I am fine using MD5 instead of SHA256. However, my decision was more based on trying to avoid collisions rather than speed. From what I read, converting to SHA256 is ~20% slower which doesn't seem an issue for this particular case. That time will be nothing compared to the time the rest of the code can take.
https://automationrhapsody.com/md5-sha-1-sha-256-sha-512-speed-performance/

Reply:
For the purpose of this homemade **temporary** and **tiny** cache, SHA-256 is good enough. The SHA vs. MD5 performance is not an issue when the queries being cached take up to 20 seconds, and clearly, the collision with less than 1,000 entries is not an issue either. In summary, SHA or hashCode are good solutions for now.